### ISSUE TITLE
Video in product details

### ISSUE THREAD

**Issue by pineportal** [smartstore/SmartStoreNET#2158]
How to add a preview image for videos
![image](https://user-images.githubusercontent.com/31403448/99423242-9a9d8a80-2915-11eb-9a9b-38bc9e819138.png)

**Comment by muratcakir** [comment]
You can't. Yo need the _MediaManager_ plugin for automatic video thumbnail generation. Although assigning thumbnails to video uploads manually is something that we'll address somewhere in the future for sure.

**Comment by pineportal** [comment]
can i create plugin for automatic video thumbnail generation? 
sample for devTools?

**Comment by muratcakir** [comment]
> can i create plugin for automatic video thumbnail generation?
> sample for devTools?

Yes, you can. `IMediaHandler` is reponsible for thumbnail generation. E.g. [ImageHandler](https://github.com/smartstore/SmartStoreNET/blob/4.x/src/Libraries/SmartStore.Services/Media/Handlers/ImageHandler.cs) creates thumbs for images. You could implement your own `VideoHandler` that inherits from `ImageHandlerBase` (which includes all thumb disk caching stuff already). Then, in `ProcessImageAsync`, you just create a thumbnail from the input video, e.g. with the help of [FFmpeg](https://ffmpeg.org/) or some other library.
